test/test_strict_encription.cpp

```diff
@@ -0,0 +1,554 @@
+/*
+ * SRT - Secure, Reliable, Transport
+ * Copyright (c) 2018 Haivision Systems Inc.
+ * 
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ * 
+ * Written by:
+ *             Haivision Systems Inc.
+ */
+
+#include <gtest/gtest.h>
+
+#include "srt.h"
+
+
+enum PEER_TYPE
+{
+    PEER_CALLER   = 0,
+    PEER_LISTENER = 1,
+    PEER_COUNT    = 2,  // Number of peers
+};
+
+
+enum CHECK_SOCKET_TYPE
+{
+    CHECK_SOCKET_CALLER   = 0,
+    CHECK_SOCKET_ACCEPTED = 1,
+    CHECK_SOCKET_COUNT    = 2,  // Number of peers
+};
+
+
+enum TEST_CASE
+{
+    TEST_CASE_A_1 = 0,
+    TEST_CASE_A_2,
+    TEST_CASE_A_3,
+    TEST_CASE_A_4,
+    TEST_CASE_A_5,
+    TEST_CASE_B_1,
+    TEST_CASE_B_2,
+    TEST_CASE_B_3,
+    TEST_CASE_B_4,
+    TEST_CASE_B_5,
+    TEST_CASE_C_1,
+    TEST_CASE_C_2,
+    TEST_CASE_C_3,
+    TEST_CASE_C_4,
+    TEST_CASE_C_5,
+    TEST_CASE_D_1,
+    TEST_CASE_D_2,
+    TEST_CASE_D_3,
+    TEST_CASE_D_4,
+    TEST_CASE_D_5,
+};
+
+
+struct TestResult
+{
+    int     connect_ret;
+    int     accept_ret;
+    int     epoll_wait_ret;
+    int     epoll_wait_error;   // error code set internally by SRT
+    int     rnum;               //< set by srt_epoll_wait
+    int     wnum;               //< set by srt_epoll_wait
+    int     socket_state[CHECK_SOCKET_COUNT];
+    int     km_state    [CHECK_SOCKET_COUNT];
+};
+
+
+struct TestCase
+{
+    bool            strictenc[PEER_COUNT];
+    std::string     password [PEER_COUNT];
```

Comment:
Does this need to be `std::string`? By the looks of it `char *` will work just as well and will avoid dynamic allocations for most of the code below.